feat(Hero): add noPaddingY and transparent background props

marcoskolodny opened the pull request:
Issue: [Link](https://jira.tid.es/browse/WEB-1782)

atabel reviewed (commented):
Checking the example playroom in the jira ticket I've seen some problems in desktop viewport:

- I still see some extra horizontal paddings that shouldn't be there:
![image](https://github.com/Telefonica/mistica-web/assets/1849576/1979a6a3-329b-4c2b-8467-b41f801f9379)

- There is a `minHeight` that we should remove when `noPaddingY` is `true`:
![image](https://github.com/Telefonica/mistica-web/assets/1849576/afe0c809-a647-479b-8c0d-ea1fcd49d21f)
![image](https://github.com/Telefonica/mistica-web/assets/1849576/44930a0a-dcb4-470e-8e5b-9297ed42edff)

Also, please include some screenshot test that cover these cases: responsive layout nesting, `noPaddingY`, background none over inverse.

marcoskolodny commented:
> Checking the example playroom in the jira ticket I've seen some problems in desktop viewport:
> 
>     * I still see some extra horizontal paddings that shouldn't be there:
>       ![image](https://private-user-images.githubusercontent.com/1849576/321531669-1979a6a3-329b-4c2b-8467-b41f801f9379.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4Mzg3ODQsIm5iZiI6MTcxMjgzODQ4NCwicGF0aCI6Ii8xODQ5NTc2LzMyMTUzMTY2OS0xOTc5YTZhMy0zMjliLTRjMmItODQ2Ny1iNDFmODAxZjkzNzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTIyODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGQ2YWZjZTcyODFlYWE3YzkwYWJhNGU3MzU0YTYxZDJkMGM5NTk1ODIyMzc4M2YyNDZhN2ZhMzc5YTAxNGIwYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.G-NyIB8uJ4mHHUb_tIIHMQE3GxJA52cPmjthy46zQD8)
> 
>     * There is a `minHeight` that we should remove when `noPaddingY` is `true`:
>       ![image](https://private-user-images.githubusercontent.com/1849576/321532180-afe0c809-a647-479b-8c0d-ea1fcd49d21f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4Mzg3ODQsIm5iZiI6MTcxMjgzODQ4NCwicGF0aCI6Ii8xODQ5NTc2LzMyMTUzMjE4MC1hZmUwYzgwOS1hNjQ3LTQ3OWItOGMwZC1lYTFmY2Q0OWQyMWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTIyODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjdlMjM2OWYxMTlkMGE4ZWVhMDg3MjZiYmU0NTQ4MTY2ZDk2MzZiNjZmMDgzZWZlZDQwMTVlZTljYjRiZDMxMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.S1A6JtNmsuf6TJOci43qnVgzs4v0u3N--6ku_jVoFVs)
>       ![image](https://private-user-images.githubusercontent.com/1849576/321532295-44930a0a-dcb4-470e-8e5b-9297ed42edff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4Mzg3ODQsIm5iZiI6MTcxMjgzODQ4NCwicGF0aCI6Ii8xODQ5NTc2LzMyMTUzMjI5NS00NDkzMGEwYS1kY2I0LTQ3MGUtOGU1Yi05Mjk3ZWQ0MmVkZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMTIyODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWI0MGEzMmQ1MjE3NTIzNTNhZGQ3NjM5YTI5NGJlM2VmMTRlM2E1ZGQxY2VmMzNjMDc3NTVkYTc0NjJjZjQzYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._h70AN1jDcHQmV8QR1nRYhlrc2fKEVl0JZxRv45FKSs)
> 
> 
> Also, please include some screenshot test that cover these cases: responsive layout nesting, `noPaddingY`, background none over inverse.

@atabel
- removed custom logic for horizontal margin and replaced it with an InternalResponsiveLayout
- removed minHeight when noPaddingY prop is used
- added screenshots to cover scenarios related to noPaddingY and nested ResponsiveLayouts (the one about background none over inverse is already covered in the skins-story).

atabel commented on src/hero.tsx:
```diff
@@ -24,13 +25,17 @@ import type {DataAttributes, RendersElement, RendersNullableElement} from './uti
+            isInverse={isInverse}
+            className={styles.layout}
+            innerDivClassName={styles.layout}
+            shouldExpandWhenNested="desktop"
```
only in desktop? why?

marcoskolodny replied:
In mobile, the picture is rendered outside the responsive layout and the content has some margin relative to the picture because of this responsive layout (expected and intended).

Without expanding (now):

![Screenshot from 2024-04-12 16-28-28](https://github.com/Telefonica/mistica-web/assets/25785151/69c7ca2a-9714-4d42-af24-afc436228631)

If we expand it (not correct):
![Screenshot from 2024-04-12 16-29-17](https://github.com/Telefonica/mistica-web/assets/25785151/a28e1289-27e5-4f9d-bd67-896d8e4b8664)

atabel replied:
I think you're wrong. See:

https://mistica-m3jn5uk7h-tuentisre.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIyALCPAYVIjQGUxqYBbGFmEmMABdmAPgA6AOwoUkACRhoIk6dIBGAQzABrAOaKArhKgBeMSAkQJMM8pUWU6qFGwSdATVvTuz9ceCeVJABJTnUdaykVKLI0MFMQakFBAAcyeAB6dLIIfViYADpDMmSSdRp8yE50gEYANgAGeoAPAE5G9IB%2BADNciWxBXOsQAKiy5IFBACV1QWwIeLr4FptIqPTxVYoAXwCORxIXGD8kAjCKQQBPcfjNWcwhkXl9w6R0050RHc3kjFnBfni6Bgf34KyiIKO-hABH6oJAXyisDIYDQ2GSs0s8QI1GwlFxFHUFBIlh0FCRKLRGKkeH61AJRJJ5xgTUE5wgVBgMAotDw5xxlDwzC0ZDBKmZgjQvmAqFKYBgtBIsDQFHWCJUqn0SUx0oAQprhBJ0NhQmgLhQuuotDARABBIRzCSvPVaw2ok0XT4BDUugAyLi0x2dBr9Ei03IwXXiAGIzCIQ1onfrLPHPZsoDN1DakqjvTAyH5gEyyLMoPAKGYOIozNs1dIkVphMkALIwHwoCBkfoO%2BKonSJUXrWxISZ55KWTv3H3qC45VmYdSo9QSQTxFz3NBkB4BOQKJSbNSaXQGIzxCxWUVReyOZyuDz7ijebBSkaBEJhCJRT9UWLxRIpNKZNkuRyoUEjFKU5SVDUDTNG09SdD0aB9AMGAXl%2BBLFBM0wYgstRLGhawbF%2BtYqHsUAHFYxzvOcVxHGYtzYPcsZPORLxvGEqZfj8wKwnRIBAhCBHnLxfhmDC-xDCRdZ5hS6LdmJ-IUPihLEq4ZIyaicmWBQNKCHSKmMoI4pshyXI8ny%2BKCmgwpCeKkrHCgsryqQSoqpxn7egagZJq6xoLmaFpWra9qWImLpGu67lRJ5yb%2Bt5vr%2BuGMCRmYMYgHG-phcG-pRSo6aCJm2bYLm%2BbAIWRnFtgpblgku7VlsUnqWQDYQM2rZPu2nZUj22B9oIaGDpErwjsU46MTAU4zpqGyvECOD4EQpBkBsIAADQgHpXB5ggADaIA6qU%2BggAAuutNJQHpZC7QAzA0q0AOy1AAHKt1T1AATAALMdWxAA

There are 2 Heros in this playroom. One inside a ResponsiveLayout and one outside it, the one inside the it should expand and be aligned as the one outside

marcoskolodny replied:
It should be fixed now :)

pladaria commented on src/__private_stories__/skin-components-story.tsx:
```diff
@@ -382,6 +383,19 @@ export const Default: StoryComponent<Args> = ({variant}) => {
                         </div>
                     </Inline>
+
+                    {/** Hero */}
```
:thinking:

aweell reviewed (commented):
Slideshow arrows seem not positioned in the right place when nesting responsive layouts?

https://tinyurl.com/2bfhv62m

yceballost commented on src/__screenshot_tests__/__image_snapshots__/hero-screenshot-test-tsx-hero-no-vertical-padding-1-snap.png:
Hero with noPadding and bullets looks broken :(

![image](https://github.com/Telefonica/mistica-web/assets/6722153/2c2ba0b2-9155-42c8-9069-a81e5825c0b1)


[Example](https://mistica-m3jn5uk7h-tuentisre.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8BlANgS1gZwBYQHcAdAOwAIyCMAXXAIQFc00ZrtSKMSA3GAJ2wxGzVtgC8wAGYBDNIIC%2BHMjRgBbccACCfPtICeAOkl8IqgBTAyLEgHNa8MgGYy8gDRkzAfXdYAHgEoyMQA%2BDyUKJAAJfghwijIAI2kwAGsbEwYSKDEiEFgZJmpcuIpVGCgMaQkS%2BKQASVVpGxga%2BLJsPjAckFxqagAHbHgAemHsCAZOmANM7H60aTwDSFVhgEYANgAGLd8ATh3hgH5JSZIaSZaQVvjF-pgwagAlaWoMCG7N%2BD3i8jb44bBGqKP5tXAwaRQTAkGASJAAFSaZGoenu3WSb14uWC0Uh0JgSGGiJswRB-zI-T4rBoLG6AAUqW9qCxfuSmbTgABtXLwmlXdw8vkAJlyAF1OX5RWT-jgwHwMP03hASN14bgMNhlJrpFZlTYyLL5Yr3uQqLQyDq0HrkTBfNRkRB2jAYGR8ARkerNQQIHwUuxrqD4rbqLo4XSFmAYPg0LA%2BGRAdK2gkGH1lXDGKmSAyMI0%2BHoyDIUjBgppHibCRnqMrs7m9KSasnMwAZLgpdMpqskFskFKuqmSboAYmx3ZSFY7ytH9cDFCgr2kmj68sbMA0lmoq7eUAcuXBJlyLkT8RwKSr-QAsuVKnSINgaCbuvKbL0DyQIHTIRVbABNOKAshKP4-ggoCpAgK4IC0Goq4IJyIDnhA3AatQ0h8CAooQVQUC0NgsGONsrhrFsQoACxSkAA)

marcoskolodny replied:
Both components are working as expected. If I put the slideshow inside a box with padding and add a blue border in Hero, you can see that the Hero's border is correct (fits the height of the content):

![MicrosoftTeams-image](https://github.com/Telefonica/mistica-web/assets/25785151/d217700d-e4c2-41bf-a6de-c07eabd1cda9)

The problem is that slideshow paints the bullets ON TOP of the content (without adding space in the bottom). In cases like Hero, if you decide to remove the bottom padding (using noPaddingY prop), it may not look good.

If, instead of Slideshow, you use a regular Carousel (in which the bullets are painted BELOW the content) and hero with noPaddingY, you will see that everything fits better:

![image](https://github.com/Telefonica/mistica-web/assets/25785151/e5ead2dc-554b-4d20-b2c3-41b1e5acf791)

marcoskolodny commented:
> Slideshow arrows seem not positioned in the right place when nesting responsive layouts?
> 
> https://tinyurl.com/2bfhv62m

After discussing offline with Alex, we've identified this is an issue with Slideshow, not related to Hero at all. I've created an issue for this: https://jira.tid.es/browse/WEB-1829

kydorn commented on src/hero.tsx:
```diff
@@ -147,11 +166,13 @@ const Hero = React.forwardRef<HTMLDivElement, HeroProps>(
                         {media}
                         <Layout isInverse={isInverse}>
-                            <Box paddingTop={24} paddingBottom={isInsideSlideShow ? 48 : 24}>
+                            <Box paddingTop={24} paddingBottom={noPaddingY ? 0 : isInsideSlideShow ? 48 : 24}>
```
if I understand this correctly, if I use noPaddingY inside a slideshow, I would need to add the pageBullets outside (or reserve this space inside the "slide", correct? It is okay, but just to make sure.

marcoskolodny replied:
Yes, that is correct :)

marcoskolodny replied:
Actually, giving it a second thought: I believe it's okay to add the bottom padding in case we put the Hero inside a slideshow (and set it to 0 otherwise if the noPaddingY prop is true). I've fixed this

marcoskolodny commented on src/hero.tsx:
```diff
@@ -123,16 +129,29 @@ const BACKGROUND_COLOR = {
     (
-        {height, background = 'default', media, desktopMediaPosition = 'left', dataAttributes, ...rest},
+        {
+            height,
```
I think we can do it in a different one, along with the refactor to try to avoid issues when using ssr

tuentisre commented:
:tada: This PR is included in version 15.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@telefonica/mistica/v/15.3.0)
- [GitHub release](https://github.com/Telefonica/mistica-web/releases/tag/v15.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
